docs/faq.md

```diff
@@ -0,0 +1,122 @@
+    });
+  }
+};
+```
+
+### But what if my package is from a Git source, and isn't on ELPA?
+
+This requires a few more lines, but it isn't by any means difficult. For an example, this installs `idris2-mode` which isn't on ELPA, but is hosted on GitHub:
+
+```nix
+programs.doom-emacs = {
+  # ...
+  emacsPackagesOverlay = self: super: {
+    idris2-mode = self.trivialBuild {
+      pname = "idris2-mode";
+      ename = "idris2-mode";
+      version = "0.0.0";
+      buildInputs = [ self.prop-menu ];
+      src = pkgs.fetchFromGitHub {
+        owner = "idris-community";
+        repo = "idris2-mode";
+        rev = "4a3f9cdb1a155da59824e39f0ac78ccf72f2ca97";
+        sha256 = "sha256-TxsGaG2fBRWWP9aas59kiNnUVD4ZdNlwwaFbM4+n81c=";
+      };
+    };
+  }
+};
+```
+
+### Help! My favourite package doesn't work after adding it in!
+
+This usually happens when the package depends on either some ELPA package or a normal package that you need to add into the `buildInputs` to make it work. 
+Let's take the above `idris2-mode` package, and I'll remove the `buildInputs` line:
+
+**WARNING**: THIS IS ERRONEOUS CODE FOR DEMONSTRATION. DO NOT USE THIS SNIPPET, THE ABOVE ONE IS THE FUNCTIONING VERSION.
+```nix
+programs.doom-emacs = {
+  # ...
+  emacsPackagesOverlay = self: super: {
+    idris2-mode = self.trivialBuild {
+      pname = "idris2-mode";
+      ename = "idris2-mode";
+      version = "0.0.0";
+      src = pkgs.fetchFromGitHub {
+        owner = "idris-community";
+        repo = "idris2-mode";
+        rev = "4a3f9cdb1a155da59824e39f0ac78ccf72f2ca97";
+        sha256 = "sha256-TxsGaG2fBRWWP9aas59kiNnUVD4ZdNlwwaFbM4+n81c=";
+      };
+    };
+  }
+};
+```
+
+This will error with (this is a part of the build log):
+
+```
+In toplevel form:
+idris2-hole-list.el:27:2: Error: Cannot open load file: No such file or directory, prop-menu
+
+In toplevel form:
+idris2-info.el:29:2: Error: Cannot open load file: No such file or directory, prop-menu
+
+In idris2-ipkg-pkgs-flags-for-current-buffer:
+idris2-ipkg-mode.el:372:2: Warning: docstring wider than 80 characters
+
+In toplevel form:
+idris2-mode.el:20:2: Error: Cannot open load file: No such file or directory, prop-menu
+
+In idris2-prover-end:
+idris2-prover.el:378:2: Warning: docstring wider than 80 characters
+
+In toplevel form:
+idris2-repl.el:29:2: Error: Cannot open load file: No such file or directory, prop-menu
+```
+
+The way to fix it is by just adding the dependency it's complaining about.
+
+## How do I enable the service?
+
+There aren't many complications about this. If you use the Home-Manager module, it's simply `services.emacs.enable = true;`. The Home-Manager module will do the rest for you.
+
+But if you're not, and you're using a standalone method, you'll need:
+
+```nix
+services.emacs = {
+  enable = true;
+  package = inputs.doom-emacs.packages.${system}.doom-emacs;
+};
+```
+
+You can now run `emacsclient -c` to connect to the daemon.
+
+## What is `trivialBuild`?
```

Comment:
If you're going this route, I think it is better to document the 4 main builders for Emacs:
- [generic](https://github.com/NixOS/nixpkgs/blob/master/pkgs/build-support/emacs/generic.nix)
- [trivialBuild](https://github.com/NixOS/nixpkgs/blob/master/pkgs/build-support/emacs/trivial.nix)
- [elpaBuild](https://github.com/NixOS/nixpkgs/blob/master/pkgs/build-support/emacs/elpa.nix)
- [melpaBuild](https://github.com/NixOS/nixpkgs/blob/master/pkgs/build-support/emacs/melpa.nix)

I think `elpaBuild` and `melpaBuild` are pretty much self-explanatory, but maybe giving an example of each builder is good.

BTW, yeah, this looks like something that should be in `nixpkgs` itself. Sadly it is not. Maybe opening an issue in `nixpkgs` and pinging the Emacs maintainers so they can document those builders :thinking: ?

Reply:
Alright! I've opened https://github.com/NixOS/nixpkgs/issues/190746. Do you think we should wait until it's documented in the Nixpkgs manual?

Reply:
To merge? Probably not.

We can always update the docs later once the documentation is generated.